packages/@aws-cdk/core/README.md

```diff
@@ -94,6 +94,24 @@ nested stack and referenced using `Fn::GetAtt "Outputs.Xxx"` from the parent.
 
 Nested stacks also support the use of Docker image and file assets.
 
+## Weak References
+
+The CDK automatically wires up cross-stack references within a single app. By default, the CDK uses the CloudFormation's
+[`Fn::ImportValue`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-importvalue.html)
+instrinsic to achieve this. To prevent interruption, CloudFormation restricts modification of any `Output` of a stack
+that is imported via this intrinsic.
+
+In many cases, this can be too restrictive. Some resources guarantee that change to its physical name does not entirely
+delete the resource. Weak references relax this restriction for specific Constructs. Instead of using the
+`Fn::ImportValue` intrinsic, [SSM Parameter
+Types](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#aws-ssm-parameter-types)
+are used instead.
+
+Enable weak reference for a resource:
+
+```ts
+resource.node.enableWeakReferences();
```

Comment:
This needs to be a tri-state to accomodate existing deployments, and you need to explain people how to migrate from a situation where  they are stuck performing an update they are planning to do.

Reply:
I'm not convinced this is needed actually or there is RoI for the added complexity.

I have only one use case for this. Users have likely worked around when they discovered this gap by keeping the layer and the function in the same stack or some other mechanism. The issue has been open long enough that I don't think anyone is waiting.
Nevertheless, I've [asked](https://github.com/aws/aws-cdk/issues/1972#issuecomment-654357523) if anyone is stuck in this state on a production stack.

Throwing this behind a feature flag that's enabled for new projects should be sufficient. Existing users will have to destroy and re-create their stacks to avail this benefit.